src/cpu.cpp

```diff
@@ -1362,6 +1362,27 @@ static int get_max_freq_khz(int cpuid)
     return max_freq_khz;
 }
 
+static int get_thread_siblings_list_count(int cpuid)
+{
+    char path[256];
+    sprintf(path, "/sys/devices/system/cpu/cpu%d/topology/thread_siblings_list", cpuid);
+
+    FILE* fp = fopen(path, "rb");
+    if (!fp)
+        return 1;
+
+    int count = 1;
+    while (!feof(fp))
+    {
+        if (fgetc(fp) == ',')
```

Comment:
The separator of `thread_siblings_list` can be `-` since kernel 5.15. My 12900k + kernel 5.19 uses hyphen.

```
What:           /sys/devices/system/cpu/cpuX/topology/core_cpus_list
Description:    human-readable list of CPUs within the same core.
                The format is like 0-3, 8-11, 14,17.
                (deprecated name: "thread_siblings_list").
Values:         decimal list.
```
https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/stable/sysfs-devices-system-cpu#L68-72
```suggestion
        char ch = fgetc(fp);
        if (ch == ',' || ch == '-')
```